### DISCUSSION TITLE
allow_population_by_field_name usage

### DISCUSSION THREAD

**Discussion by FlorianLudwig** [pydantic/pydantic#3277] [Question] [answered]
Using `pydantic==1.8.2` I cannot get `allow_population_by_field_name` working. Example:

```
from typing import Optional
import pydantic

class Test(pydantic.BaseModel):
    foo: Optional[str] = pydantic.Field(alias="bar")
    class Config:
        allow_population_by_field_name: True
```

I would except both, setting foo and bar to work like this:
```
>>> Test(foo=1)
Test(foo=None)
>>> Test(bar=1)
Test(foo='1')
```

It seems like the option doesn't have any effect.

**Comment by PrettyWood** [comment, marked as answer]
Hi @FlorianLudwig 
It's `allow_population_by_field_name = True` (not `:`)

**Reply by FlorianLudwig** [reply]
:facepalm: 

thanks!
